test/e2e/httpproxy/query_parameter_condition_match_test.go

```diff
@@ -0,0 +1,349 @@
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-prefix-ignorecase",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:       "targetPrefixIgnoreCase",
+									Prefix:     "prefixval",
+									IgnoreCase: true,
+								},
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-suffix",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:   "targetSuffix",
+									Suffix: "ffixValue",
+								},
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-suffix-ignorecase",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:       "targetSuffixIgnoreCase",
+									Suffix:     "ffixvalueignorecase",
+									IgnoreCase: true,
+								},
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-regex",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:  "targetRegex",
+									Regex: "^RegexV.*",
+								},
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-contains",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:     "targetContains",
+									Contains: "nsVal",
+								},
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-contains-ignorecase",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:       "targetContainsIgnoreCase",
+									Contains:   "svalueIgnorec",
+									IgnoreCase: true,
+								},
+							},
+						},
+					},
+					{
+						Services: []contourv1.Service{
+							{
+								Name: "echo-query-parameter-present",
+								Port: 80,
+							},
+						},
+						Conditions: []contourv1.MatchCondition{
+							{
+								QueryParameter: &contourv1.QueryParameterMatchCondition{
+									Name:    "targetPresent",
+									Present: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		f.CreateHTTPProxyAndWaitFor(p, e2e.HTTPProxyValid)
+
+		type scenario struct {
+			queryParams    map[string]string
+			expectResponse int
+			expectService  string
+		}
+
+		cases := []scenario{
```

Comment:
similar to the gw api testing, would be good to have coverage to demonstrate the case where a request has multiple query params with the same key so we have it defined how contour will behave, e.g. https://github.com/projectcontour/contour/blob/052a12d88decdb068926333b9f90ae77cb650a5c/test/e2e/gateway/query_param_match_test.go#L69-L73